### PR TITLE
feat: Add Stack Pack for Next.js

### PR DESCRIPTION
**Pull request by kyliau**
This commit adds a Lighthouse Stack Pack for Next.js based on
documentation and recommendations on Next.js website.

Example:

Screen Shot 2021-11-16 at 10 00 54 AM

### CODE REVIEW

**Review by paulirish — APPROVED** [review]
all the proposed text looks good to me. succinct and direct.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -10,5 +10,5 @@ jobs:
         with:
           node-version: 14.x
 
-      - run: npm install
+      - run: npm install --ignore-scripts
```
Comment (kyliau) [inline .github/workflows/ci.yml:13]: During npm install, the Lighthouse package which is installed from GitHub
(@master branch) triggers the newly added `prepack` script [1], but due to
a bug in NPM [2] the devDependencies are not available when the script
is run, thus making the installation fail.

[1]: GoogleChrome/lighthouse#13261
[2]: npm/cli#1865

File: packs/next.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+const UIStrings = {
+  'unused-css-rules': 'PurgeCSS can be used in Next.js to remove unused rules from stylesheets. [Learn more](https://purgecss.com/guides/next.html).',
+  'modern-image-formats': 'Use Next.js Image Optimization API to serve modern formats like WebP and AVIF. [Learn more](https://nextjs.org/docs/api-reference/next/image#acceptable-formats).',
```
Comment (connorjclark) [inline packs/next.js:6]: We'll need the `/** ...` strings as found in other packs, as they are presented to our translators for context. I am about to update `general.js` with the template to use for each of these, here is the PR #68

Can you put the terms that should not be translated in backticks?

File: packs/next.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  'offscreen-images': 'Use Next.js Image component, `next/image`, which defaults to `loading="lazy"`. [Learn more](https://nextjs.org/docs/api-reference/next/image#loading).',
+  'render-blocking-resources': 'Use Next.js Script component, `next/script`, to defer loading of non-critical third-party scripts. [Learn more](https://nextjs.org/docs/basic-features/script).',
```
Comment (connorjclark) [inline packs/next.js:8]:
```suggestion
  'render-blocking-resources': 'Use the `next/script` component to defer loading of non-critical third-party scripts. [Learn more](https://nextjs.org/docs/basic-features/script).',
```

this is relevant to all entries:

1) adding definite article `the`
2) Drop "x component, \`y\`" and prefer "\`y\` component", which avoids potential translation confusion. I am not sure if component should be translated, so I left it out of the backticks. Do next.js tech writers have any preference?

File: packs/next.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  'unused-javascript': 'Use Webpack Bundle Analyzer to detect unused JavaScript code. [Learn mode](https://github.com/vercel/next.js/tree/canary/packages/next-bundle-analyzer)',
+  'uses-long-cache-ttl': 'Make sure the right strategy is used for immutable assets and SSR. [Learn more](https://nextjs.org/docs/going-to-production#caching).',
+  'uses-optimized-images': 'Use Next.js Image component, `next/image`, to optimize images. [Learn more](https://nextjs.org/docs/basic-features/image-optimization).',
+  'uses-text-compression': 'Make sure compression is enabled on your Next.js server. [Learn more](https://nextjs.org/docs/api-reference/next.config.js/compression).',
+  'uses-responsive-images': 'Use Next.js Image component, `next/image` to set the appropriate `sizes`. [Learn more](https://nextjs.org/docs/api-reference/next/image#sizes).',
+  'user-timings': 'Consider using Next.js Analytics to measure your app\'s real-world performance. [Learn more](https://nextjs.org/docs/advanced-features/measuring-performance).',
```
Comment (connorjclark) [inline packs/next.js:14]:
```suggestion
  'user-timings': 'Consider using `Next.js` Analytics to measure your app\'s real-world performance. [Learn more](https://nextjs.org/docs/advanced-features/measuring-performance).',
```

File: packs/next.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+const UIStrings = {
+  'unused-css-rules': 'PurgeCSS can be used in Next.js to remove unused rules from stylesheets. [Learn more](https://purgecss.com/guides/next.html).',
+  'modern-image-formats': 'Use Next.js Image Optimization API to serve modern formats like WebP and AVIF. [Learn more](https://nextjs.org/docs/api-reference/next/image#acceptable-formats).',
+  'offscreen-images': 'Use Next.js Image component, `next/image`, which defaults to `loading="lazy"`. [Learn more](https://nextjs.org/docs/api-reference/next/image#loading).',
+  'render-blocking-resources': 'Use Next.js Script component, `next/script`, to defer loading of non-critical third-party scripts. [Learn more](https://nextjs.org/docs/basic-features/script).',
+  'unused-javascript': 'Use Webpack Bundle Analyzer to detect unused JavaScript code. [Learn mode](https://github.com/vercel/next.js/tree/canary/packages/next-bundle-analyzer)',
+  'uses-long-cache-ttl': 'Make sure the right strategy is used for immutable assets and SSR. [Learn more](https://nextjs.org/docs/going-to-production#caching).',
```
Comment (connorjclark) [inline packs/next.js:10]:
```suggestion
  'uses-long-cache-ttl': 'Configure caching for immutable assets and SSR. [Learn more](https://nextjs.org/docs/going-to-production#caching).',
```

Comment (connorjclark) [reply packs/next.js:10]: ("Make sure" probably won't translate naturally)

**Review by connorjclark — CHANGES_REQUESTED** [review]
Thanks for the PR, I'm sure Next.js developers will find this helpful!

File: packs/next.js [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
...='0 0 207 124'%3E%3Cg fill='%23000' fill-rule='nonzero'%3E%3Cpath d='M48.942 32.632h38.96v3.082H52.512v23.193h33.278v3.082H52.513v25.464h35.794v3.081H48.942V32.632Zm42.45 0h4.139l18.343 25.464 18.749-25.464L158.124.287l-41.896 60.485 21.59 29.762h-4.302l-19.642-27.086L94.15 90.534h-4.22l21.751-29.762-20.29-28.14Zm47.967 3.082v-3.082h44.397v3.082h-20.453v54.82h-3.571v-54.82h-20.373ZM.203 32.632h4.464l61.557 91.671-25.439-33.769L3.936 37.011l-.162 53.523H.203zM183.397 86.523c.738 0 1.276-.563 1.276-1.29 0-.727-.538-1.29-1.276-1.29-.73 0-1.277.563-1.277 1.29 0 .727.547 1.29 1.277 1.29Zm3.509-3.393c0 2.146 1.555 3.549 3.822 3.549 2.414 0 3.874-1.446 3.874-3.956v-8.837h-1.946v8.828c0 1.394-.704 2.138-1.946 2.138-1.112 0-1.867-.692-1.893-1.722h-1.911Zm10.24-.113c.14 2.233 2.007 3.662 4.787 3.662 2.97 0 4.83-1.498 4.83-3.887 0-1.878-1.06-2.917-3.632-3.514l-1.38-.338c-1.634-.38-2.294-.891-2.294-1.783 0-1.125 1.025-1.86 2.563-1.86 1.459 0 2.466.718 2.649 1.869h1.893c-.113-2.103-1.971-3.583-4.516-3.583-2.737 0-4.56 1.48-4.56 3.704 0 1.835 1.033 2.926 3.3 3.454l1.616.39c1.659.389 2.388.96 2.388 1.912 0 1.108-1.146 1.913-2.71 1.913-1.676 0-2.84-.753-3.005-1.939h-1.928Z'/%3E%3C/g%3E%3C/svg%3E`;
```
Comment (connorjclark) [inline packs/next.js:2]:
```suggestion
const icon = `data:image/svg+xml,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 207 124"><path d="M48.942 32.632h38.96v3.082h-35.39v23.193H85.79v3.082H52.513v25.464h35.794v3.081H48.942V32.632Zm42.45 0h4.139l18.343 25.464 18.749-25.464L158.124.287l-41.896 60.485 21.59 29.762h-4.302l-19.642-27.086L94.15 90.534h-4.22l21.751-29.762-20.29-28.14Zm47.967 3.082v-3.082h44.397v3.082h-20.453v54.82h-3.571v-54.82h-20.373ZM.203 32.632h4.464l61.557 91.671-25.439-33.769L3.936 37.011l-.162 53.523H.203zm183.194 53.891c.738 0 1.276-.563 1.276-1.29 0-.727-.538-1.29-1.276-1.29-.73 0-1.277.563-1.277 1.29 0 .727.547 1.29 1.277 1.29Zm3.509-3.393c0 2.146 1.555 3.549 3.822 3.549 2.414 0 3.874-1.446 3.874-3.956v-8.837h-1.946v8.828c0 1.394-.704 2.138-1.946 2.138-1.112 0-1.867-.692-1.893-1.722h-1.911Zm10.24-.113c.14 2.233 2.007 3.662 4.787 3.662 2.97 0 4.83-1.498 4.83-3.887 0-1.878-1.06-2.917-3.632-3.514l-1.38-.338c-1.634-.38-2.294-.891-2.294-1.783 0-1.125 1.025-1.86 2.563-1.86 1.459 0 2.466.718 2.649 1.869h1.893c-.113-2.103-1.971-3.583-4.516-3.583-2.737 0-4.56 1.48-4.56 3.704 0 1.835 1.033 2.926 3.3 3.454l1.616.39c1.659.389 2.388.96 2.388 1.912 0 1.108-1.146 1.913-2.71 1.913-1.676 0-2.84-.753-3.005-1.939h-1.928Z" fill="%23000"/></svg>`;
```

Minified with https://jakearchibald.github.io/svgomg/ . also FYI no need to encode brackets in data urls.

Comment (kyliau) [reply packs/next.js:2]: https://github.com/GoogleChrome/lighthouse-stack-packs/blob/399f81b2ae98012264748aeb5c673069a3a7a247/CONTRIBUTING.md?plain=1#L36

**Review by housseindjirdeh — APPROVED** [review]
This looks great, thanks @kyliau! Excited to see a Next.js stack pack become a reality :)

File: general.js [hunk 7]
```diff
@@ -26,6 +26,8 @@ const UIStrings = {
 }
 
 module.exports = {
+  id: '<please fill>',
+  title: '<please fill>',
```
[the remark this answers is not in the document]
Comment (kyliau) [reply general.js:30]: `general.js` is a template, so yes, this is a TODO. Connor will fix this file in a separate PR.

**Comment by kyliau** [comment]
@connorjclark could you please do another pass? All comments have been addressed. Thank you 😃

**Review by connorjclark — APPROVED** [review]

**Pull request merged**